src/libcollections/string.rs

```diff
@@ -809,14 +986,23 @@ impl String {
         self.len() == 0
     }
 
-    /// Truncates the string, returning it to 0 length.
+    /// Truncates this `String`, removing all contents.
+    ///
+    /// While this means the `String` will have a length of zero, it does not
+    /// touch its capacity.
     ///
     /// # Examples
     ///
+    /// Basic usage:
+    ///
     /// ```
-    /// let mut s = "foo".to_string();
+    /// let mut s = String::from("foo");
```

Comment:
I can see how this aligns better in the `from_utf16` example above, but this documentation is setting an example of how to create strings in the wild. In general although `String::from` is possible I would say the more idiomatic way to do this is `foo.to_string()`, so it seems a little odd to me that all these examples are switching to `String::from`?